TribalWars/src/Infrastructure/Repository/PlayerRepository.cs

```diff
@@ -1,20 +1,38 @@
-﻿using ApplicationCore.Interfaces.Repository;
-using Infrastructure.Data;
-using Microsoft.EntityFrameworkCore;
 using System;
 using System.Threading.Tasks;
+using ApplicationCore.Entities;
+using ApplicationCore.Interfaces.Repository;
+using Infrastructure.Data;
+using Microsoft.EntityFrameworkCore;
 
 namespace Infrastructure.Repository
 {
-    public class PlayerRepository : BaseRepository, IPlayerRepository
+    public class PlayerRepository : IPlayerRepository
     {
-        public PlayerRepository(ApplicationDbContext context) : base(context)
+        private readonly ApplicationDbContext _context;
+
+        public PlayerRepository(ApplicationDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<Player> GetByUserId(Guid userId)
         {
+            return await _context.Players.FirstOrDefaultAsync(p => p.Id == userId);
         }
 
         public async Task<bool> PlayerExistById(Guid playerId)
         {
-            return await Context.Players.AnyAsync(p => p.Id == playerId);
+            return await _context.Players.AnyAsync(p => p.Id == playerId);
+        }
+
+        public async Task<bool> AddAsync(Player player)
+        {
+            _context.Players.Add(player);
+            return await _context.SaveChangesAsync() > 0;
```

Comment:
What will happen here if SaveChangesAsync() throws an exception? That's usually the most common fail scenario.
Maybe you can handle it in BaseRepository?